src/Compilers/CSharp/Portable/Symbols/Extensions/SynthesizedExtensionThisParameterMetadataSymbol.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SynthesizedExtensionThisParameterMetadataSymbol : ParameterSymbol
+    {
+        private readonly Symbol _containingSymbol;
+
+        internal SynthesizedExtensionThisParameterMetadataSymbol(SourceExtensionMetadataMethodSymbol forMethod)
+        {
+            _containingSymbol = forMethod;
+        }
+
+        internal SynthesizedExtensionThisParameterMetadataSymbol(SourceExtensionMetadataPropertySymbol forProperty)
+        {
+            _containingSymbol = forProperty;
+        }
+
+        // PROTOTYPE(roles): Confirm what name should we use for this parameter
+        //                   Is this name going to hurt or help EE when this parameter is lifted into a closure?
+        public override string Name => GeneratedNames.ThisProxyFieldName();
+
+        public override bool IsDiscard => false;
+
+        public override TypeWithAnnotations TypeWithAnnotations
+            => TypeWithAnnotations.Create(
+                    ((SourceExtensionTypeSymbol)_containingSymbol.ContainingSymbol).GetExtendedTypeNoUseSiteDiagnostics(null),
```

Comment:
Just want to confirm, from our discussion earlier this week, we'll want to change this to the extension type at some point (instead of the extended type) and we'll let erasure deal with it.

Reply:
> Just want to confirm, from our discussion earlier this week, we'll want to change this to the extension type at some point (instead of the extended type) and we'll let erasure deal with it.

While that approach probably is going to work, at the moment I am not sure what would be an advantage of doing that.

Reply:
Though, I think it will be worth following up once the "erase" functionality is there.